src/chatd.cpp

```diff
@@ -5270,8 +5410,7 @@ void Chat::onAddReaction(Id msgId, Id userId, std::string reaction)
 
 void Chat::onDelReaction(Id msgId, Id userId, std::string reaction)
 {
-    Idx messageIdx = msgIndexFromId(msgId);
-    if (messageIdx == CHATD_IDX_INVALID)
+    if (mDbInterface->getIdxOfMsgidFromHistory(msgId) == CHATD_IDX_INVALID)
```

Comment:
Same than for `onAddReaction()`. Not efficient to look up at DB for every reaction just to check if the `msgid` is locally available or not. Please, if really needed, find a more efficient way to check it.